### ISSUE TITLE
Halo 4 support

### ISSUE THREAD

**Issue by ELREVENGE** [ELREVENGE/HaloMCCPCSaveTransferTool#7]
**Is your feature request related to a problem? Please describe.**
Halo 4 files are not currently supported by the tool.

**Describe the solution you'd like**
Support for Halo 4 files to be read and transferred with the tool.

**Describe alternatives you've considered**
Horizon method might work but I haven't been able to test it yet.

**Additional context**
The FileInfo class has as case for Halo 4 but I haven't been able to test them yet. I'm moving so if someone can get me a Halo 4 map to test with that would help speed things up. I currently have 2 game type files I had been trying to read which I can get the name and description of. The game sometimes stores map and game type info in their files differently so I need to test with a map as well.

**Comment by 17mcalpal** [comment]
Here are some maps and game modes off of my 360. I a 98% sure these are the Halo 4. Just would have clicked the wrong folder. Good luck!
[Halo 4 maps and gamemodes.zip](https://github.com/ELREVENGE/HaloMCCPCSaveTransferTool/files/5576663/Halo.4.maps.and.gamemodes.zip)

**Comment by ELREVENGE** [comment]
> Here are some maps and game modes off of my 360. I a 98% sure these are the Halo 4. Just would have clicked the wrong folder. Good luck!
> [Halo 4 maps and gamemodes.zip](https://github.com/ELREVENGE/HaloMCCPCSaveTransferTool/files/5576663/Halo.4.maps.and.gamemodes.zip)

I'm looking through the files and if I'm right the gametypes are stored the same way maps are so I might be able to push out a fix on the dev branch. I'll start actually coding on my laptop tomorrow. If all goes well expect a new dev branch push and build that I'll link in a new comment on this issue tomorrow.

**Comment by ELREVENGE** [comment]
![image](https://user-images.githubusercontent.com/2909708/100020134-2ab75600-2d94-11eb-811a-8ae31b847c49.png)

I've got it reading and displaying all of the files you sent. I'm setting up the export window code now so it drops the files off in the correct place when the export button is hit.

**Comment by ELREVENGE** [comment]
I just pushed the changes to the dev branch and compiled the program to be tested. I don't have the game downloaded on my laptop currently so if someone tests this before I get the chance to download the game and test it myself please comment here. Once it's been tested I'll push the changes to the master branch and create a new release for everyone to download.
[Halo.MCC.PC.Save.Transfer.Tool_v.2.0.1.0_NoInstaller.zip](https://github.com/ELREVENGE/HaloMCCPCSaveTransferTool/files/5586239/Halo.MCC.PC.Save.Transfer.Tool_v.2.0.1.0_NoInstaller.zip)
[Halo.MCC.PC.Save.Transfer.Tool_v.2.0.1.0_Installer.zip](https://github.com/ELREVENGE/HaloMCCPCSaveTransferTool/files/5586240/Halo.MCC.PC.Save.Transfer.Tool_v.2.0.1.0_Installer.zip)

**Comment by ELREVENGE** [comment]
I've tested it and it seems to work so I'll merge the dev into the master and create a new release. **Just a heads up I noticed that some files (specifically the map "battle world") didn't seem to appear in game.** I've seen this before and an issue was created and closed see issue #3 for more details. The solution to the issue is to swap which directory the file is in, so if its in the private directory and doesn't show up move it to the built in directory and it should work.
